doc/adr/0016- end-to-end testing-strategy.rst

```diff
@@ -0,0 +1,199 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Cli flags that require use of plutus scripts E.g. tx-out-reference-script-file orcalculate-plutus-script-cost
+* Checking cli behaviour when script evaluation passes. This could be displaying expected fee correctly.
+* Checking cli behaviour when script evaluation fails. This can be how different types errors are formatted. 
+
+Argument
+--------
+
+Pros of building and maintaining our own test framework
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Plutus tools team will have full ownership of the end-to-end test environment and its priorities
+  for ``plutus``.
+* Plutus scripts can be defined alongside the tests. ``cardano-node-tests`` requires pre-compiled scripts.
+* Tests will demonstrate how these Haskell packages can be used together to guide plutus application
+  development using the node apis. Particularly useful for less experienced haskell developers.  
+* Possible to define tests once and run at different levels. For example, on private or public
+  testnets and with ``cardano-node-emulator`` emulated node.
+* Benefits from use of all ``plutus`` apis. For example, using PlutusTx to produce scripts using a
+  typed interface, and optionally the Contract monad from ``plutus-apps``.
+* Have the oportunity to add more componenets under test at a later stage, such as Marconi or a PAB.
+* ``cardano-cli`` would not be a dependency for plutus test coverage so no risk of being blocked by its
+  stage of development.
+* Plutus team can implement and review majority of tests in Haskell rather than Python, which is
+  likely to be the team's preference. Also won't need to review as many tests in ``cardano-node-tests``.
+* Less friction caused by cross-team: planning, dependencies and expectations. Plutus team won't
+  need to wait for node test team to implement the tests. It's likely that other node/cli features
+  will often be prioritised.
+* This approach will will improve our high-level perspective of each componenent and help guide
+  UX improvements.
+* Node team are not pressured to focus on plutus scenarios, they retain control of their priorities.
+
+Cons of building and maintaining our own test framework
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* ``cardano-node-tests`` is well established and already has useful features such as: running tests in
+  different eras, transitioning between eras, reporting, and measuring deviations in script cost.
+* We are proposing to add another means of spinning up a local testnet when there are already
+  multiple others (see notes section below).
+* No major work needed to continue using ``cardano-node-tests``. Node test team will continue to
+  upgrade/maintain it regardless.
+* Some coverage of plutus tests already exist and can continue to be added in
+  ``cardano-node-tests`` by the node qa and plutus teams.
+* Going forwards, many plutus tests will be mostly a copy/paste job. For example, the existing
```

Comment:
I would this point by maybe talking the problem of test duplication?

Reply:
I'm actually trying to explain how low maintenance it would be to cover `plutus` tests going forwards with the python framework. I'll try emphasising that.

Reply:
I understand what you mean now. I can actually reword as a positive point for why duplication isn't so bad.